cmd/download_test.go

```diff
@@ -77,6 +77,65 @@ func TestDownloadWithoutFlags(t *testing.T) {
 	}
 }
 
+func TestSolutionFile(t *testing.T) {
+	testCases := []struct {
+		name, file, expectedPath, expectedURL string
+	}{
+		{
+			name:         "filename with special character",
+			file:         "special-char-filename#.txt",
+			expectedPath: "special-char-filename#.txt",
+			expectedURL:  "http://www.example.com/special-char-filename%23.txt",
+		},
+		{
+			name:         "filename with leading slash",
+			file:         "/with-leading-slash.txt",
+			expectedPath: filepath.Join("/", "with-leading-slash.txt"),
```

Comment:
On Linux and Mac, `filepath.Join("/", "with-leading-slash.txt")` returns `/with-leading-slash.txt`.
On Windows, `filepath.Join("/", "with-leading-slash.txt")` returns `\with-leading-slash.txt`.

I think this would be less confusing, if we said:

```
fmt.Sprintf("%s%s", os.PathSeparator, "with-leading-slash.txt")
```

Likewise in the other tests, I think we should be very explicit about where we expect a forward slash backward slash or encoded slash or an os-specific filepath separator.

Reply:
How about `filepath.Join(string(os.PathSeparator), "with-leading-slash.txt")`?

Reply:
`filepath.Join` inserts the `os.PathSeparator`. In this case, I don't think using `filepath.Join` expresses clearly enough what would actually be the expected path, because it's not clear when it will translate a slash, when it will add another slash, etc. Looking at your suggested line of code I would wonder if it ended up with two slashes, for example.

Reply:
@kytrinyx I have used fmt.Sprintf instead of filrpath.Join. Does this look better?